app/app.rb

```diff
@@ -277,12 +277,14 @@ class App < Sinatra::Base
 					end
 				end
 			elsif params[:file]
-				file.parent.upload(params[:file], file.name)
+				if params[:file].is_a?(Hash) && params[:file][:tempfile]
```

Comment:
No because then it leaves it open to fall through to the parent else, which we don't want in this case.